File: algoritmos_e_estruturas/sort/busca_binaria.py

def busca_binaria(lista:list[int], target:int):
    
    baixo = 0
    alto = len(lista) - 1
    while baixo <= alto:
        meio = (baixo + alto) // 2
        chute = lista[meio]
        if chute == target:
            return chute
        if chute > target:
            alto = meio - 1
        else:
            baixo = meio + 1 
    return None      

def main():
    lista = [num for num in range(10) if num % 2 != 0]
    print(lista)
    print(busca_binaria(lista, 3))
    
main()

File: algoritmos_e_estruturas/Sorting/Binary_Search/busca_binaria.cpp
#include <iostream>

using namespace std;

int busca_binaria(int lista[], int tam, int target){
    int baixo = 0;
    int alto = tam -1;
    int meio, chute;

    while (baixo <= alto){
        meio = (baixo + alto) / 2;
        chute = lista[meio];
        if (chute == target)
            return chute; // retorna o endereco do item
        if (chute > target)
            alto = meio - 1;
        else
            baixo = meio + 1;
    }
    return -1;
}

int main(){

    int lista[5] = {2,4,6,8,10};
    int tam = 5;


    while (true){

        cout << 
        cout << "Lista: [ ";
        for (int i=0; i<5; i++){
            cout << lista[i] << " " ;
        }
        cout << "]" << endl;

        cout << busca_binaria(lista, tam, 3);
    }

}
